hz.ps1

```diff
@@ -981,6 +1038,40 @@ if ($doTagVersion)
     }
 }
 
+# generate codecs
+if ($doCodecs) {
```

Comment:
the idea of codec generator is to not modify the codecs after generation.
This block of script still borrow some of the code generator responsibility.
I see a normalization stage below, if there is such need, it should be in the generator.

Reply:
I get it